api/src/main/java/jakarta/data/metamodel/BasicAttribute.java

```diff
@@ -42,9 +42,9 @@ public interface BasicAttribute<T,V> extends Attribute<T>, Expression<T,V> {
     static <T,V> BasicAttribute<T,V> of(Class<T> entityClass,
                                         String name,
                                         Class<V> attributeType) {
-        Objects.requireNonNull(entityClass, "entity class is required");
-        Objects.requireNonNull(name, "entity attribute name is required");
-        Objects.requireNonNull(attributeType, "entity attribute type is required");
+        Objects.requireNonNull(entityClass, "Entity class is required");
+        Objects.requireNonNull(name, "Entity attribute name is required");
```

Comment:
But @njr-11, once the parameter is "name," it should match with the variable, right?

I mean, I passed the name parameter as `null`, and I received the "Entity attribute name is required" sounds odd.

Reply:
Maybe we could have a wrapper class to handle this message, such as:

```java
Validation. requireNonNull(name, "name);
```

Where can we centralize the error message?

Reply:
@njr-11 that you be nice.

One more question, about all the messages be the same:

Like:

```java
 Objects.requireNonNull(entityClass, "Entity class is required");
 Objects.requireNonNull(name, "Entity attribute name is required");
```

should it be this instead:


```java
 Objects.requireNonNull(entityClass, "entityClass class is required");
 Objects.requireNonNull(name, "name attribute name is required");
```

I mean, I believe the message name should match the parameter name, right?

Reply:
It definitely should not be `"name attribute name is required"`. That doesn't make any sense. From your other example, `"entityClass class is required"`, I think you are asking to ensure we have the exact variable name in the message rather than splitting it into separate words.  So `entityClass` rather than `Entity class`.  I'm fine with that.  In order to match the consistent style of beginning with a capital letter, we could have something like, `"The entityClass argument is required."`